src/crimson/os/seastore/object_data_handler.cc

```diff
@@ -310,66 +313,106 @@ overwrite_ops_t prepare_ops_list(
   }
 
   // prepare to insert
-  extent_to_remap_list_t to_remap;
-  for (auto &region : to_write) {
-    if (region.is_data()) {
-      visitted++;
-      assert(region.to_write.has_value());
-      int erased_num = 0;
-      if (pre_alloc_addr_removed.contains(region.addr, region.len) &&
-	  region.len <= delta_based_overwrite_max_extent_size) {
-	erased_num = std::erase_if(
-	  ops.to_remove,
-	  [&region, &to_remap](auto &r) {
-	    interval_set<uint64_t> range;
-	    range.insert(r->get_key(), r->get_length());
-	    if (range.contains(region.addr, region.len)) {
-	      to_remap.push_back(extent_to_remap_t::create_overwrite(
-		0, region.len, r->get_key(), r->get_length(), *region.to_write));
-	      return true;
+  return seastar::do_with(
+    visitted,
+    pre_alloc_addr_removed,
+    pre_alloc_addr_remapped,
+    ctx,
+    extent_to_remap_list_t(),
+    std::move(ops),
+    [&to_write, do_size=delta_based_overwrite_max_extent_size]
+    (auto &visitted, auto &pre_alloc_addr_removed,
+     auto &pre_alloc_addr_remapped, auto &ctx, auto &to_remap, auto &ops) {
+    return trans_intr::do_for_each(
+      to_write,
+      [&ops, &visitted, &pre_alloc_addr_removed, &pre_alloc_addr_remapped,
+       &ctx, do_size, &to_remap](auto &region) {
+      if (region.is_data()) {
+	visitted++;
+	assert(region.to_write.has_value());
+	auto do_if_not_clones = [&ctx, &region](
+	  std::function<TransactionManager::get_pin_iertr::future<>()> &&f) {
+	  return ctx.tm.get_pins(ctx.t, region.addr, region.len
```

Comment:
Related to tracking the original pins from the above comments.

As far as I can see, the pin containing the region.addr~len should be already available from ops.to_remove/to_remap, which means that `TM::get_pins()` is not needed here to lookup the lba-tree again.

What do you think?

Reply:
Correct. Sorry for my fault.